{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../plink-182/data/filtered.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ../plink-182/data/filtered\n",
      "  --vcf ../plink-182/data/filtered.vcf.gz\n",
      "\n",
      "385411 MB RAM detected; reserving 192705 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: 79k variants complete.\r",
      "--vcf: 80k variants complete.\r",
      "--vcf: 81k variants complete.\r",
      "--vcf: 82k variants complete.\r",
      "--vcf: 83k variants complete.\r",
      "--vcf: 84k variants complete.\r",
      "--vcf: 85k variants complete.\r",
      "--vcf: 86k variants complete.\r",
      "--vcf: 87k variants complete.\r",
      "--vcf: 88k variants complete.\r",
      "--vcf: 89k variants complete.\r",
      "--vcf: 90k variants complete.\r",
      "--vcf: 91k variants complete.\r",
      "--vcf: 92k variants complete.\r",
      "--vcf: 93k variants complete.\r",
      "--vcf: 94k variants complete.\r",
      "--vcf: 95k variants complete.\r",
      "--vcf: 96k variants complete.\r",
      "--vcf: 97k variants complete.\r",
      "--vcf: 98k variants complete.\r",
      "--vcf: 99k variants complete.\r",
      "--vcf: 100k variants complete.\r",
      "--vcf: 101k variants complete.\r",
      "--vcf: 102k variants complete.\r",
      "--vcf: 103k variants complete.\r",
      "--vcf: 104k variants complete.\r",
      "--vcf: 105k variants complete.\r",
      "--vcf: 106k variants complete.\r",
      "--vcf: 107k variants complete.\r",
      "--vcf: 108k variants complete.\r",
      "--vcf: 109k variants complete.\r",
      "--vcf: 110k variants complete.\r",
      "--vcf: 111k variants complete.\r",
      "--vcf: 112k variants complete.\r",
      "--vcf: 113k variants complete.\r",
      "--vcf: 114k variants complete.\r",
      "--vcf: 115k variants complete.\r",
      "--vcf: 116k variants complete.\r",
      "--vcf: 117k variants complete.\r",
      "--vcf: 118k variants complete.\r",
      "--vcf: 119k variants complete.\r",
      "--vcf: 120k variants complete.\r",
      "--vcf: 121k variants complete.\r",
      "--vcf: 122k variants complete.\r",
      "--vcf: 123k variants complete.\r",
      "--vcf: 124k variants complete.\r",
      "--vcf: 125k variants complete.\r",
      "--vcf: 126k variants complete.\r",
      "--vcf: 127k variants complete.\r",
      "--vcf: 128k variants complete.\r",
      "--vcf: 129k variants complete.\r",
      "--vcf: 130k variants complete.\r",
      "--vcf: 131k variants complete.\r",
      "--vcf: 132k variants complete.\r",
      "--vcf: 133k variants complete.\r",
      "--vcf: 134k variants complete.\r",
      "--vcf: 135k variants complete.\r",
      "--vcf: 136k variants complete.\r",
      "--vcf: 137k variants complete.\r",
      "--vcf: 138k variants complete.\r",
      "--vcf: 139k variants complete.\r",
      "--vcf: 140k variants complete.\r",
      "--vcf: 141k variants complete.\r",
      "--vcf: 142k variants complete.\r",
      "--vcf: 143k variants complete.\r",
      "--vcf: 144k variants complete.\r",
      "--vcf: 145k variants complete.\r",
      "--vcf: 146k variants complete.\r",
      "--vcf: 147k variants complete.\r",
      "--vcf: 148k variants complete.\r",
      "--vcf: 149k variants complete.\r",
      "--vcf: 150k variants complete.\r",
      "--vcf: 151k variants complete.\r",
      "--vcf: 152k variants complete.\r",
      "--vcf: 153k variants complete.\r",
      "--vcf: 154k variants complete.\r",
      "--vcf: 155k variants complete.\r",
      "--vcf: 156k variants complete.\r",
      "--vcf: 157k variants complete.\r",
      "--vcf: 158k variants complete.\r",
      "--vcf: 159k variants complete.\r",
      "--vcf: 160k variants complete.\r",
      "--vcf: 161k variants complete.\r",
      "--vcf: 162k variants complete.\r",
      "--vcf: 163k variants complete.\r",
      "--vcf: 164k variants complete.\r",
      "--vcf: 165k variants complete.\r",
      "--vcf: 166k variants complete.\r",
      "--vcf: 167k variants complete.\r",
      "--vcf: 168k variants complete.\r",
      "--vcf: 169k variants complete.\r",
      "--vcf: 170k variants complete.\r",
      "--vcf: ../plink-182/data/filtered-temporary.bed +\n",
      "../plink-182/data/filtered-temporary.bim +\n",
      "../plink-182/data/filtered-temporary.fam written.\n",
      "170073 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ../plink-182/data/filtered.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "170073 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ../plink-182/data/filtered.bed + ../plink-182/data/filtered.bim +\n",
      "../plink-182/data/filtered.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "plink --vcf ../../plink-182/data/filtered.vcf.gz --make-bed --out ../../plink-182/data/filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../Project/Runs/Noenvironmenteffect/Nocontrol/results.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../plink-182/data/filtered\n",
      "  --linear\n",
      "  --out ../Project/Runs/Noenvironmenteffect/Nocontrol/results\n",
      "  --pheno ../Project/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1.phen\n",
      "\n",
      "385411 MB RAM detected; reserving 192705 MB for main workspace.\n",
      "170073 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../Project/Runs/Noenvironmenteffect/Nocontrol/results.nosex .\n",
      "2504 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "170073 variants and 2504 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "../Project/Runs/Noenvironmenteffect/Nocontrol/results.assoc.linear ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ../Project/Runs/Yesenvironmenteffect/Nocontrol/results.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ../plink-182/data/filtered\n",
      "  --linear\n",
      "  --out ../Project/Runs/Yesenvironmenteffect/Nocontrol/results\n",
      "  --pheno ../Project/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1.phen\n",
      "\n",
      "385411 MB RAM detected; reserving 192705 MB for main workspace.\n",
      "170073 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "../Project/Runs/Yesenvironmenteffect/Nocontrol/results.nosex .\n",
      "2504 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "170073 variants and 2504 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to\n",
      "../Project/Runs/Yesenvironmenteffect/Nocontrol/results.assoc.linear ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Plink results when we don't account for pop strat\n",
    "\n",
    "DATADIR=../../plink-182/data\n",
    "#DATADIR=~/ps3/\n",
    "PREFIX=${DATADIR}/filtered # use for plink --bfile\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "TYPE=Nocontrol\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results # use for plink --out\n",
    "\n",
    "# YOUR CODE HERE\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --allow-no-sex --out ${OUTPREFIX1}\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "# Change to more useable output format\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is terminated.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "## Plink results if we use PCA\n",
    "\n",
    "TYPE=PCA_HPS_whiten_False\n",
    "DATADIR=../../plink-182/data\n",
    "PREFIX=${DATADIR}/filtered # use for plink --bfile\n",
    "USERPREFIX1='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "USERPREFIX2='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}\n",
    "\n",
    "\n",
    "\n",
    "# YOUR CODE HERE\n",
    "### NUMVARS1\n",
    "NUMVARS=1\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1 --allow-no-sex --out ${OUTPREFIX1}\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1 --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "\n",
    "### NUMVARS2\n",
    "NUMVARS=2\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX1}\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "\n",
    "#### NUMVARS3\n",
    "#NUMVARS=3\n",
    "#OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX1}\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX2}\n",
    "#\n",
    "#cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "#cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "#\n",
    "#\n",
    "#### NUMVARS4\n",
    "#NUMVARS=4\n",
    "#OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX1}\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX2}\n",
    "#\n",
    "#cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "#cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "#\n",
    "#\n",
    "#### NUMVARS5\n",
    "#NUMVARS=5\n",
    "#OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX1}\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX2}\n",
    "#\n",
    "#cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "#cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "#\n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is terminated.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "## Plink results if we use PCoA\n",
    "\n",
    "TYPE=PCoA_HPS_metric_manhattan\n",
    "DATADIR=../../plink-182/data\n",
    "PREFIX=${DATADIR}/filtered # use for plink --bfile\n",
    "USERPREFIX1='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "USERPREFIX2='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}\n",
    "\n",
    "mkdir ../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}\n",
    "mkdir ../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}\n",
    "\n",
    "# YOUR CODE HERE\n",
    "### NUMVARS1\n",
    "NUMVARS=1\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1 --allow-no-sex --out ${OUTPREFIX1}\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1 --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "\n",
    "### NUMVARS2\n",
    "NUMVARS=2\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX1}\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "\n",
    "### NUMVARS3\n",
    "#NUMVARS=3\n",
    "#OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX1}\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "#cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "#cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "\n",
    "### NUMVARS4\n",
    "#NUMVARS=4\n",
    "#OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX1}\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "#cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "#cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "\n",
    "### NUMVARS5\n",
    "#NUMVARS=5\n",
    "#OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "#OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${NUMVARS} # use for plink --out\n",
    "\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX1}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX1}\n",
    "#plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX2}.phen --covar ${COVARFILE}.cov --covar-number 1-${NUMVARS} --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "#cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "#cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "###### Using TSNE braycurtis\n",
    "\n",
    "TYPE=t-SNE_HPS_metric_braycurtis_perplexity\n",
    "DATADIR=../../plink-182/data\n",
    "PREFIX=${DATADIR}/filtered # use for plink --bfile\n",
    "\n",
    "\n",
    "#mkdir ../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}\n",
    "#mkdir ../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}\n",
    "\n",
    "# YOUR CODE HERE\n",
    "### PERPLEX2\n",
    "PERPLEX=2\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "### PERPLEX3\n",
    "PERPLEX=3\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "### PERPLEX10\n",
    "PERPLEX=10\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "### PERPLEX30\n",
    "PERPLEX=30\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "### PERPLEX100\n",
    "PERPLEX=100\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "###### Using TSNE manhattan\n",
    "\n",
    "TYPE=t-SNE_HPS_metric_manhattan_perplexity\n",
    "DATADIR=../../plink-182/data\n",
    "PREFIX=${DATADIR}/filtered # use for plink --bfile\n",
    "\n",
    "#mkdir ../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}\n",
    "#mkdir ../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}\n",
    "\n",
    "# YOUR CODE HERE\n",
    "### PERPLEX2\n",
    "PERPLEX=2\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "### PERPLEX3\n",
    "PERPLEX=3\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "### PERPLEX10\n",
    "PERPLEX=10\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "### PERPLEX30\n",
    "PERPLEX=30\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "### PERPLEX100\n",
    "PERPLEX=100\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PERPLEX}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PERPLEX} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "######## Using UMAP\n",
    "\n",
    "TYPE=UMAP_HPS\n",
    "DATADIR=../../plink-182/data\n",
    "PREFIX=${DATADIR}/filtered # use for plink --bfile\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "\n",
    "#mkdir ../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}\n",
    "#mkdir ../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}\n",
    "\n",
    "\n",
    "\n",
    "# YOUR CODE HERE\n",
    "### PARAMS UMAP_HPS_metric_braycurtis_min_dist_0.1_n_neighbors_10\n",
    "PARAMS=metric_braycurtis_min_dist_0.1_n_neighbors_10\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PARAMS}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PARAMS} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PARAMS} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "\n",
    "### PARAMS UMAP_HPS_metric_manhattan_min_dist_0.1_n_neighbors_10\n",
    "PARAMS=metric_manhattan_min_dist_0.1_n_neighbors_10\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PARAMS}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PARAMS} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PARAMS} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n",
    "\n",
    "### PARAMS UMAP_HPS_metric_canberra_min_dist_0.1_n_neighbors_50\n",
    "PARAMS=metric_canberra_min_dist_0.1_n_neighbors_50\n",
    "COVARFILE=../../plink-182/cov_dr_outputs/${TYPE}_${PARAMS}\n",
    "OUTPREFIX1=../GWASminiProject/Runs/Noenvironmenteffect/${TYPE}/results_${PARAMS} # use for plink --out\n",
    "OUTPREFIX2=../GWASminiProject/Runs/Yesenvironmenteffect/${TYPE}/results_${PARAMS} # use for plink --out\n",
    "\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Noenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov --allow-no-sex --out ${OUTPREFIX1}\n",
    "USERPREFIX='../GWASminiProject/SimulatedPhenotypes/Yesenvironmenteffect/pheno_sim1'\n",
    "plink --bfile ${PREFIX} --linear --pheno ${USERPREFIX}.phen --covar ${COVARFILE}.cov  --allow-no-sex --out ${OUTPREFIX2}\n",
    "\n",
    "cat ${OUTPREFIX1}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX1}.assoc.linear.tab\n",
    "cat ${OUTPREFIX2}.assoc.linear | sed -r 's/^\\s+//g' | sed -r 's/\\s+/\\t/g' > ${OUTPREFIX2}.assoc.linear.tab\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
